plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java

```diff
@@ -1346,15 +1346,23 @@ private String getBuildImageId(ProgressStatus progressStatus) {
 
     protected <T> T parseResponseStreamAndClose(InputStream inputStream, Class<T> clazz) throws IOException {
         try (InputStreamReader reader = new InputStreamReader(inputStream)) {
-            return GSON.fromJson(reader, clazz);
+            T objectFromJson = GSON.fromJson(reader, clazz);
+            if (objectFromJson == null) {
+                throw new IOException("Docker responded with an empty stream.");
```

Comment:
Empty stream says nothing to a user I suppose. And in certain cases, e.g. 301 or 201, an empty body is OK, so this message can be frustrating a bit. 
WDYT about next code instead:
```
LOG.error("Docker response doesn't contain any data, though it was expected to contain some. Stack trace: {}", Thread.currentThread().getStackTrace());
throw new IOException("Internal server error. Unexpected response body from Docker daemon received.");
```